flink-formats/flink-orc/src/main/java/org/apache/flink/orc/OrcFilters.java

```diff
@@ -0,0 +1,372 @@
+					.put(BuiltInFunctionDefinitions.GREATER_THAN, createGreaterThanPredicateConverter())
+					.put(BuiltInFunctionDefinitions.GREATER_THAN_OR_EQUAL, createGreaterThanEqualsPredicateConverter())
+					.put(BuiltInFunctionDefinitions.LESS_THAN, createLessThanPredicateConverter())
+					.put(BuiltInFunctionDefinitions.LESS_THAN_OR_EQUAL, createLessThanEqualsPredicateConverter())
+					.build();
+
+	private static boolean isUnaryValid(CallExpression callExpression) {
+		return callExpression.getChildren().size() == 1 && callExpression.getChildren().get(0) instanceof FieldReferenceExpression;
+	}
+
+	private static boolean isBinaryValid(CallExpression callExpression) {
+		return callExpression.getChildren().size() == 2 && ((callExpression.getChildren().get(0) instanceof FieldReferenceExpression && callExpression.getChildren().get(1) instanceof ValueLiteralExpression) ||
+				(callExpression.getChildren().get(0) instanceof ValueLiteralExpression && callExpression.getChildren().get(1) instanceof FieldReferenceExpression));
+	}
+
+	private static Tuple2<String, PredicateLeaf.Type> getTuple2Args(CallExpression callExp){
+		if (!isUnaryValid(callExp)) {
+			// not a valid predicate
+			LOG.debug("Unsupported predicate [{}] cannot be pushed into OrcFileSystemFormatFactory.", callExp);
+			return null;
+		}
+
+		PredicateLeaf.Type colType = toOrcType(((FieldReferenceExpression) callExp.getChildren().get(0)).getOutputDataType());
+		if (colType == null) {
+			// unsupported type
+			LOG.debug("Unsupported predicate [{}] cannot be pushed into OrcFileSystemFormatFactory.", callExp);
+			return null;
+		}
+
+		String colName = getColumnName(callExp);
+
+		return Tuple2.of(colName, colType);
+	}
+
+	private static Tuple3<String, PredicateLeaf.Type, Serializable> getTuple3Args(CallExpression callExp){
+		if (!isBinaryValid(callExp)) {
+			// not a valid predicate
+			LOG.debug("Unsupported predicate [{}] cannot be pushed into OrcFileSystemFormatFactory.", callExp);
+			return null;
+		}
+
+		PredicateLeaf.Type litType = getLiteralType(callExp);
+		if (litType == null) {
+			// unsupported literal type
+			LOG.debug("Unsupported predicate [{}] cannot be pushed into OrcFileSystemFormatFactory.", callExp);
+			return null;
+		}
+
+		String colName = getColumnName(callExp);
+
+		// fetch literal and ensure it is serializable
+		Object literalObj = getLiteral(callExp).get();
+		Serializable literal;
+		// validate that literal is serializable
+		if (literalObj instanceof Serializable) {
+			literal = (Serializable) literalObj;
+		} else {
+			LOG.warn("Encountered a non-serializable literal of type {}. " +
+							"Cannot push predicate [{}] into OrcFileSystemFormatFactory. " +
+							"This is a bug and should be reported.",
+					literalObj.getClass().getCanonicalName(), callExp);
+			return null;
+		}
+
+		return Tuple3.of(colName, litType, literal);
+	}
+
+	private static Function<CallExpression, OrcSplitReader.Predicate> createIsNullPredicateConverter(){
```

Comment:
You can create a method: `convertIsNull`, and:
`.put(BuiltInFunctionDefinitions.IS_NULL, OrcFilters::convertIsNull`.
Others can be the same.